commit 76707e34f47707d8937a3a8400f980f2f304637c
Author: Naftali Goldstein <naftali.goldstein@intel.com>
Date:   Sun Jul 15 15:26:27 2018 +0300

    iwlwifi: fw: do not set sgi bits for HE connection
    
    [ Upstream commit 5c2dbebb446539eb9640bf59a02756d6e7f1fc53 ]
    
    If the association supports HE, HT/VHT rates will never be used for Tx
    and therefore there's no need to set the sgi-per-channel-width-support
    bits, so don't set them in this case.
    
    Fixes: 110b32f065f3 ("iwlwifi: mvm: rs: add basic implementation of the new RS API handlers")
    Signed-off-by: Naftali Goldstein <naftali.goldstein@intel.com>
    Signed-off-by: Luca Coelho <luciano.coelho@intel.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c b/drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c
index 8169d1450b3b..d1c1a8069c7e 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/rs-fw.c
@@ -98,8 +98,12 @@ static u8 rs_fw_sgi_cw_support(struct ieee80211_sta *sta)
 {
 	struct ieee80211_sta_ht_cap *ht_cap = &sta->ht_cap;
 	struct ieee80211_sta_vht_cap *vht_cap = &sta->vht_cap;
+	struct ieee80211_sta_he_cap *he_cap = &sta->he_cap;
 	u8 supp = 0;
 
+	if (he_cap && he_cap->has_he)
+		return 0;
+
 	if (ht_cap->cap & IEEE80211_HT_CAP_SGI_20)
 		supp |= BIT(IWL_TLC_MNG_CH_WIDTH_20MHZ);
 	if (ht_cap->cap & IEEE80211_HT_CAP_SGI_40)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1029_76707e34f47707d8937a3a8400f980f2f304637c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1029_76707e34f47707d8937a3a8400f980f2f304637c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
